lti_consumer/lti_xblock.py

```diff
@@ -830,41 +828,37 @@ def is_past_due(self):
             close_date = due_date
         return close_date is not None and timezone.now() > close_date
 
-    def _get_lti1p1_consumer(self):
+    def _get_lti_consumer(self):
         """
-        Returns a preconfigured LTI 1.1 consumer.
+        Returns a preconfigured LTI consumer depending on the value.
 
         If the block is configured to use LTI 1.1, set up a
         base LTI 1.1 consumer class.
-        This class does NOT store state between calls.
-        """
-        key, secret = self.lti_provider_key_secret
-        return LtiConsumer1p1(self.launch_url, key, secret)
-
-    def _get_lti1p3_consumer(self):
-        """
-        Returns a preconfigured LTI 1.3 consumer.
 
         If the block is configured to use LTI 1.3, set up a
         base LTI 1.3 consumer class with all block related
         configuration services.
 
+        This uses the LTI API to fetch the configuration
+        from the models and instance the LTI client.
+
         This class does NOT store state between calls.
         """
-        return LtiConsumer1p3(
-            iss=get_lms_base(),
-            lti_oidc_url=self.lti_1p3_oidc_url,
-            lti_launch_url=self.lti_1p3_launch_url,
-            client_id=self.lti_1p3_client_id,
-            deployment_id="1",
-            # XBlock Private RSA Key
-            rsa_key=self.lti_1p3_block_key,
-            rsa_key_id=self.lti_1p3_client_id,
-            # LTI 1.3 Tool key/keyset url
-            tool_key=self.lti_1p3_tool_public_key,
-            tool_keyset_url=None,
+        # Runtime import since this will only run in the
+        # Open edX LMS/Studio environments.
+        # pylint: disable=import-outside-toplevel
+        from lti_consumer.api import (
+            get_or_create_local_lti_config,
+            get_lti_consumer
+        )
+
+        lti_config_id = get_or_create_local_lti_config(
```

Comment:
Instead of having to get/create the `lti_config_id` and then immediately looking up the model by id, how about having something like `get_lti_consumer(config_id=None, block=self)`, which will get_or_create the config and return the consumer? And since configuration is only on the block for now, you can avoid having to load the block again (in `_load_block`), since it's already being passed in.